presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -360,7 +361,12 @@ public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHan
         IcebergTableHandle icebergTableHandle = (IcebergTableHandle) tableHandle;
         IcebergColumnHandle columnHandle = (IcebergColumnHandle) source;
         Table icebergTable = getIcebergTable(session, icebergTableHandle.getSchemaTableName());
-        icebergTable.updateSchema().renameColumn(columnHandle.getName(), target).commit();
+        Transaction transaction = icebergTable.newTransaction();
+        transaction.updateSchema().renameColumn(columnHandle.getName(), target).commit();
```

Comment:
Curious to know why are we changing direct updateSchema to updateSchema through transaction? I see updateSchema through transaction maintains an update list so is it to maintain the order of schema & spec update?

Reply:
From my understanding, it's not about the ordering of the operations, it's that you want to update both the partition spec and the schema update in the same transaction. If you commit the schema update without the partition update or vice-verse, you'll end up with a table in an inconsistent state.

By performing both updates on the transaction, after calling `Transaction#commit` you'll be guaranteed that either both operations succeed or both fail.